src/ripple/shamap/SHAMap.h

```diff
@@ -424,7 +452,11 @@ inline
 SHAMap::const_iterator&
 SHAMap::const_iterator::operator++()
 {
-    item_ = map_->peekNextItem(item_->key(), stack_);
+    auto temp = map_->peekNextItem(item_->key(), stack_);
+    if (temp)
+        item_ = temp->peekItem().get();
+    else
+        item_ = nullptr;
```

Comment:
Yes.  I've thought about making them bidirectional.  It is technically very doable.  But we haven't had a motivating use case yet, and I don't want to add code we don't need.